### PR TITLE
Golang idioms

### PR DESCRIPTION
**Pull request by jmacd**
I was browsing and made some suggestions, which I will annotate.

### CODE REVIEW

File: pkg/otel/metrics/otlp_to_arrow.go [hunk 1]
```diff
@@ -88,10 +84,9 @@ func OtlpMetricsToArrowRecords(rr *air.RecordRepository, request *collogspb.Expo
 func addMetric(rr *air.RecordRepository, resMetrics *metricspb.ResourceMetrics, scopeMetrics *metricspb.ScopeMetrics, metricName string, dataPoints []*metricspb.NumberDataPoint, config *MultivariateMetricsConfig) error {
 	if mvKey, ok := config.Metrics[metricName]; ok {
 		return multivariateMetric(rr, resMetrics, scopeMetrics, dataPoints, mvKey)
-	} else {
-		univariateMetric(rr, resMetrics, scopeMetrics, metricName, dataPoints)
-		return nil
 	}
+	univariateMetric(rr, resMetrics, scopeMetrics, metricName, dataPoints)
```
Comment (jmacd) [inline pkg/otel/metrics/otlp_to_arrow.go:88]: The standard `golint` settings in OTel will require this de-indenting blocks after unconditional returns.

File: pkg/otel/metrics/otlp_to_arrow.go [hunk 2]
```diff
@@ -100,21 +95,18 @@ func multivariateMetric(rr *air.RecordRepository, resMetrics *metricspb.Resource
 
 	for _, ndp := range dataPoints {
 		sig := DataPointSig(ndp, multivariateKey)
-		newEntry := false
 		stringSig := string(sig)
 		record := records[stringSig]
 
+		var multivariateMetricName *string
```
Comment (jmacd) [inline pkg/otel/metrics/otlp_to_arrow.go:101]: (This is just a readability fix.)

File: pkg/otel/metrics/otlp_to_arrow.go [hunk 3]
```diff
@@ -141,15 +133,14 @@ func multivariateMetric(rr *air.RecordRepository, resMetrics *metricspb.Resource
 		}
 
 		if multivariateMetricName == nil {
-			emptyString := ""
-			multivariateMetricName = &emptyString
+			multivariateMetricName = new(string)
```
Comment (jmacd) [inline pkg/otel/metrics/otlp_to_arrow.go:136]: I almost never use `new`, but it has a few uses.

File: pkg/otel/metrics/otlp_to_arrow.go [hunk 4]
```diff
@@ -183,23 +174,22 @@ func univariateMetric(rr *air.RecordRepository, resMetrics *metricspb.ResourceMe
 			record.U64Field(constants.START_TIME_UNIX_NANO, ndp.StartTimeUnixNano)
 		}
 
-		attributes := common.NewAttributes(ndp.Attributes)
-		if attributes != nil {
+		if attributes := common.NewAttributes(ndp.Attributes); attributes != nil {
```
Comment (jmacd) [inline pkg/otel/metrics/otlp_to_arrow.go:177]: This avoids entering `attributes` into the scope.

File: pkg/otel/metrics/otlp_to_arrow.go [hunk 5]
```diff
@@ -42,14 +43,14 @@ func OtlpMetricsToArrowRecords(rr *air.RecordRepository, request *collogspb.Expo
 		for _, scopeMetrics := range resourceMetrics.ScopeMetrics {
 			for _, metric := range scopeMetrics.Metrics {
 				if metric.Data != nil {
-					switch metric.Data.(type) {
+					switch t := metric.Data.(type) {
```
[the remark this answers is not in the document]
Comment (lquerel) [reply pkg/otel/metrics/otlp_to_arrow.go:46]: done

**Comment by lquerel** [comment]
Thanks, I will apply these idioms to the rest of the code.

**Pull request closed without merging**